model/task_queue_service_dependency.go

```diff
@@ -244,8 +244,8 @@ func (d *basicCachedDAGDispatcherImpl) FindNextTask(ctx context.Context, spec Ta
 	// If the host just ran a task group, give it one back.
 	if spec.Group != "" {
 		taskGroupID := compositeGroupID(spec.Group, spec.BuildVariant, spec.Project, spec.Version)
-		taskGroupUnit, ok := d.getTaskGroup(taskGroupID) // schedulableUnit
-		if ok {
+		taskGroupUnit, hasDispatchableTask := d.getTaskGroup(taskGroupID)
+		if hasDispatchableTask {
 			if next := d.tryMarkNextTaskGroupTaskDispatched(taskGroupUnit); next != nil {
 				return next
 			}
```

Comment:
Can we update the fall through comment below this to say either the task group is not present, _or_ none of the remaining tasks in the group are dispatchable, so it's safe to move on from this task group?